src/pybind/mgr/dashboard/frontend/src/app/shared/services/password-policy.service.ts

```diff
@@ -1,26 +1,76 @@
 import { Injectable } from '@angular/core';
 
 import { I18n } from '@ngx-translate/i18n-polyfill';
+import { forkJoin, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { SettingsService } from '../api/settings.service';
 
 @Injectable({
   providedIn: 'root'
 })
 export class PasswordPolicyService {
-  constructor(private i18n: I18n) {}
+  constructor(private i18n: I18n, private settingsService: SettingsService) {}
 
-  getHelpText() {
-    return this.i18n(
-      'Required rules for password complexity:\n\
-    - must contain at least 8 characters\n\
-    - cannot contain username\n\
-    - cannot contain any keyword used in Ceph\n\
-    - cannot contain any repetitive characters e.g. "aaa"\n\
-    - cannot contain any sequential characters e.g. "abc"\n\
-    - must consist of characters from the following groups:\n\
-      * alphabetic a-z, A-Z\n\
-      * numbers 0-9\n\
-      * special chars: !"#$%& \'()*+,-./:;<=>?@[\\]^_`{{|}}~\n\
-      * any other characters (signs)'
+  getHelpText(): Observable<string> {
+    const observables = [];
```

Comment:
Just a suggestion, it might be more readable to use an object instead of an array of observables:
```
   const observables = {
      enabled: this.settingsService.getValue('PWD_POLICY_ENABLED'),
      length: this.settingsService.getValue('PWD_POLICY_CHECK_LENGTH_ENABLED'),
      min_length: this.settingsService.getValue('PWD_POLICY_MIN_LENGTH'),
      oldpwd: this.settingsService.getValue('PWD_POLICY_CHECK_OLDPWD_ENABLED'),
      username: this.settingsService.getValue('PWD_POLICY_CHECK_USERNAME_ENABLED'),
      exclusion: this.settingsService.getValue('PWD_POLICY_CHECK_EXCLUSION_LIST_ENABLED'),
      repetitive: this.settingsService.getValue('PWD_POLICY_CHECK_REPETITIVE_CHARS_ENABLED'),
      sequential: this.settingsService.getValue('PWD_POLICY_CHECK_SEQUENTIAL_CHARS_ENABLED'),
      complexity: this.settingsService.getValue('PWD_POLICY_CHECK_COMPLEXITY_ENABLED')
    };
```
then you can access its values with `resp['enabled']` or `resp.enabled`

Reply:
Great, thanks for this hint. That will make it much easier to read the code.